src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -166,4 +185,62 @@ private JsonArray listTemplates(String pageToken, JsonArray jsonElements) throws
     }
     return jsonElements;
   }
+
+  public List<StructuredRecord> listTemplateData(String templateName) throws IOException, AribaException,
```

Comment:
you have already catch an AribaException below, do we really need this in throws

Reply:
yes because it is needed in the further methods, mentioned after the catch block.